core/templates/dev/head/pages/suggestion_editor/ShowSuggestionModalForEditorView.js

```diff
@@ -52,6 +56,40 @@ oppia.controller('ShowSuggestionModalForEditorView', [
 
     $scope.currentContent = currentContent;
     $scope.newContent = newContent;
+    $scope.oldContent = $scope.newContent;
+    $scope.editedContent = $scope.newContent;
+    $scope.commitMessage = description;
+    $scope.reviewMessage = null;
+
+    $scope.isEditButtonShow = function() {
+      return $scope.editButtonShown;
+    };
+
+    $scope.isSuggestionEditorIsShown = function() {
+      return $scope.suggestionEditorShown;
+    };
+
+    $scope.isSaveButtonShown = function() {
+      return (!$scope.editButtonShown &&
+        $scope.suggestionEditorShown);
+    };
+
+    $scope.editSuggestion = function() {
+      $scope.editButtonShown = false;
+      $scope.suggestionEditorShown = true;
+    };
+
+    $scope.saveSuggestion = function() {
+      $scope.editButtonShown = true;
+      $scope.suggestionEditorShown = false;
+      $uibModalInstance.close({
+        action: SuggestionModalService.ACTION_EDIT_SUGGESTION,
+        newSuggestionHtml: $scope.editedContent,
+        summaryMessage: $scope.summaryMessage,
+        suggestionType: $scope.suggestionType,
+        oldContent: $scope.oldContent
+      });
+    };
```

Comment:
Will you be using the above functions in the editor view?

Reply:
Hello @nithusha21, can you elaborate more on this question?
I don't get it what you want to ask.